File: examples/EngScript_examples/PHP/EngScriptTest.php


function engScript_randomStringGenerator(){
    return;
    (array(1,3,5,3)[rand((0 - 1)+1, count(array(1,3,5,3))-1)]);
}
print(rand(2+1, 5-1))
print(rand(2+1, 7-1))
print(rand(2+1, 5-1))
if($engScript_a){
    b;
}
elseif($engScript_v){
    d;
}
else {
    d;
}
//"The next line uses the string concatenation operator"
$engScript_str1 . $engScript_str2
//"The next line uses the array concatenation operator"
array_merge($engScript_arr1, $engScript_arr2)
function engScript_openingTag($engScript_theTag){
    return "<" . $engScript_theTag . ">";
}
function engScript_closingTag($engScript_theTag){
    return "</" . $engScript_theTag . ">";
}
function engScript_surroundWithTag($engScript_theString, $engScript_theTag){
    return engScript_openingTag($engScript_theTag) . $engScript_theString . engScript_closingTag($engScript_theTag);
}
function engScript_boldText($engScript_str1){
    return engScript_surroundWithTag($engScript_str1,"b");
}
function engScript_italicText($engScript_str1){
    return engScript_surroundWithTag($engScript_str1,"i");
}
function engScript_boldWikiText($engScript_str1){
    return "'''" . $engScript_str1 . "'''";
}
function engScript_italicWikiText($engScript_str1){
    return "''" . $engScript_str1 . "''";
}
function engScript_wikiLink($engScript_pageTitle, $engScript_linkText){
    return "[[" . $engScript_pageTitle . "|" . $engScript_linkText . "]]";
}
function engScript_markdownLink($engScript_theURL, $engScript_linkText){
    return "[" . $engScript_linkText . "](" . $engScript_theURL . ")";
}
strlen("hello")
substr($engScript_a,2,5)
"derp" . "herp";
(3 > 4)
(3 < 4)
(3 >= 4)
(3 <= 4)
print($engScript_lawz)
foreach ($engScript_b as $engScript_a){
    derp
}
//"Hello World!"
"common " . " lisp"
function engScript_helloWorld($engScript_a, $engScript_b){
    if($engScript_a){
        return (3 - (((1 * 1) + 2) / 4));
    }
    elseif($engScript_b){
        return 2;
    }
    else {
        return 3;
    }
}

File: examples/EngScript_examples/REBOL/REBOL projects/other files/EngScriptTest.r
[comment This code is written in REBOL 3.]
[comment This code is written in REBOL 3.]
engScript_a [
    b
]
engScript_v [
    d
]
true [
    d
]
[comment "The next line uses the string concatenation operator"]
append engScript_str1 engScript_str2
[comment "The next line uses the array concatenation operator"]
append engScript_arr1 engScript_arr2
engScript_openingTag: func [engScript_theTag [string!]] [
    return append "<" append engScript_theTag ">"
]
engScript_closingTag: func [engScript_theTag [string!]] [
    return append "</" append engScript_theTag ">"
]
engScript_surroundWithTag: func [engScript_theString [string!], engScript_theTag [string!]] [
    return append engScript_openingTag engScript_theTag append engScript_theString engScript_closingTag engScript_theTag
]
engScript_boldText: func [engScript_str1 [string!]] [
    return engScript_surroundWithTag engScript_str1 "b"
]
engScript_italicText: func [engScript_str1 [string!]] [
    return engScript_surroundWithTag engScript_str1 "i"
]
engScript_boldWikiText: func [engScript_str1 [string!]] [
    return append "'''" append engScript_str1 "'''"
]
engScript_italicWikiText: func [engScript_str1 [string!]] [
    return append "''" append engScript_str1 "''"
]
engScript_wikiLink: func [engScript_pageTitle [string!], engScript_linkText [string!]] [
    return append "[[" append engScript_pageTitle append "|" append engScript_linkText "]]"
]
engScript_markdownLink: func [engScript_theURL [string!], engScript_linkText [string!]] [
    return append "[" append engScript_linkText append "](" append engScript_theURL ")"
]
length? "hello"
copy/part skip engScript_a 2 5
append "derp" "herp"
(3 > 4)
(3 < 4)
(3 >= 4)
(3 <= 4)
print engScript_lawz
foreach engScript_a engScript_b [
    derp
]
[comment "Hello World!"]
append "common " " lisp"
engScript_helloWorld: func [engScript_a [integer!], engScript_b [integer!]] [
    case [
    engScript_a [
        return (3 - (((1 * 1) + 2) / 4))
    ]
    engScript_b [
        return 2
    ]
    true [
        return 3
    ]
    ]
]